netmiko/extreme/extreme_ssh.py

```diff
@@ -1,13 +1,123 @@
+        Handle prompt changes during command executions.
+        """
+        if len(args) >= 1:
+            command_string = args[0]
+
+        # command that prompt a question
+        if command_string in QUESTIONS.keys():
+            output = self.answer_questions(command_string)
+
+        # all other commands
+        else:
+            # Turn off 'auto_find_prompt' which loses the striped base prompt
+            kwargs.update({'auto_find_prompt': False})
+            # refresh prompt, which could have been changed on previous command
+            self.set_base_prompt()
+            output = super(ExtremeSSH, self).send_command(*args, **kwargs)
+
+        return output
+
+    def send_config_set(self, config_commands=None, exit_config_mode=True, delay_factor=1,
+                        max_loops=150, strip_prompt=False, strip_command=False):
+        """
+        Overwrite send_config_set to acomodate for commands which return a question
+        and expect a reply.
+        """
+        delay_factor = self.select_delay_factor(delay_factor)
+        if config_commands is None:
+            return ''
+        if not hasattr(config_commands, '__iter__'):
+            raise ValueError("Invalid argument passed into send_config_set")
+
+        # Send config commands
+        output = ''
+        for command_string in config_commands:
+
+            # command that prompt a question
+            if command_string in QUESTIONS.keys():
+                output += self.answer_questions(command_string)
+
+            # all other commands
+            else:
+                self.write_channel(self.normalize_cmd(command_string))
+                time.sleep(delay_factor * .5)
+
+        # Gather output
+        output += self._read_channel_timing(delay_factor=delay_factor, max_loops=max_loops)
+
+        output = self._sanitize_output(output)
+        return output
+
+    def set_base_prompt(self, *args, **kwargs):
+        """
+        Extreme attaches an id to the prompt. The id increases with every command.
+        It needs to br stripped off to match the prompt. Eg.
+
+            testhost.1 #
+            testhost.2 #
+            testhost.3 #
+
+        If new config is loaded and not saved yet, a '* ' prefix appears before the
+        prompt, eg.
+
+            * testhost.4 #
+            * testhost.5 #
+        """
+        cur_base_prompt = super(ExtremeSSH, self).set_base_prompt(*args, **kwargs)
+        match = re.search(r'[\*\s]{0,2}(.*)\.\d+', cur_base_prompt)
+        if match:
+            # strip off .\d from base_prompt
+            self.base_prompt = match.group(1)
+            return self.base_prompt
+
+    def config_mode(self, config_command=''):
+        """No configuration mode on Extreme."""
+        return ''
+
+    def check_config_mode(self, check_string='#'):
+        """Checks whether in configuration mode. Returns a boolean."""
+        return super(ExtremeSSH, self).check_config_mode(check_string=check_string)
```

Comment:
Are we just checking if in `enable_mode` here? If so probably should just call `self.check_enable_mode()`. This will make the code clearer.